lib/TippyTopProvider.js

```diff
@@ -0,0 +1,67 @@
+/* globals require, JSON */
+"use strict";
+
+const data = require("sdk/self").data;
+const URL = require("sdk/url").URL;
+
+const DEFAULT_OPTIONS = {
+  sites: null
+};
+
+function TippyTopProvider(options = {}) {
+  this.options = Object.assign({}, DEFAULT_OPTIONS, options);
+  this.init();
+}
+
+TippyTopProvider.prototype = {
+  /**
+   * Initialize the sites object
+   */
+  init() {
+    // If the sites were passed into the constructor (for testability), use those.
+    // Else, read them from top_sites.json.
+    let sites;
+    if (this.options.sites) {
+      sites = this.options.sites;
+    } else {
+      sites = JSON.parse(data.load("content/favicons/top_sites.json"));
```

Comment:
Not sure if you want to touch the JSON file directly vs the `require("tippy-top-sites")` exported version via [index.js](https://github.com/nchapman/tippy-top-sites/blob/master/index.js) (since we inject a couple values on the fly). /cc @k88hudson